iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/BusAdapter.java

```diff
@@ -32,189 +33,190 @@ protected BusAdapter(final Bus delegate, final MergingViewIndex index) {
+    public Iterable<LccConverterStation> getLccConverterStations() {
+        return getLccConverterStationStream().collect(Collectors.toSet());
     }
 
     @Override
-    public Stream<Line> getLineStream() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public Stream<LccConverterStation> getLccConverterStationStream() {
+        return getDelegate().getLccConverterStationStream().map(getIndex()::getLccConverterStation);
     }
 
     @Override
-    public Iterable<ThreeWindingsTransformer> getThreeWindingsTransformers() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public Iterable<VscConverterStation> getVscConverterStations() {
+        return getVscConverterStationStream().collect(Collectors.toSet());
     }
 
     @Override
-    public Stream<ThreeWindingsTransformer> getThreeWindingsTransformerStream() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public Stream<VscConverterStation> getVscConverterStationStream() {
+        return getDelegate().getVscConverterStationStream().map(getIndex()::getVscConverterStation);
     }
 
+    // -------------------------------
+    // Simple delegated methods ------
+    // -------------------------------
     @Override
-    public Iterable<Generator> getGenerators() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public VoltageLevelAdapter getVoltageLevel() {
+        return getIndex().getVoltageLevel(getDelegate().getVoltageLevel());
     }
 
     @Override
-    public Stream<Generator> getGeneratorStream() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public double getV() {
+        return getDelegate().getV();
     }
 
     @Override
-    public Iterable<Battery> getBatteries() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public BusAdapter setV(final double v) {
+        getDelegate().setV(v);
+        return this;
     }
 
     @Override
-    public Stream<Battery> getBatteryStream() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public double getAngle() {
+        return getDelegate().getAngle();
     }
 
     @Override
-    public Iterable<Load> getLoads() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public BusAdapter setAngle(final double angle) {
+        getDelegate().setAngle(angle);
+        return this;
     }
 
     @Override
-    public Stream<Load> getLoadStream() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public double getP() {
+        return getDelegate().getP();
     }
 
     @Override
-    public Iterable<ShuntCompensator> getShuntCompensators() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public double getQ() {
+        return getDelegate().getQ();
     }
 
     @Override
-    public Stream<ShuntCompensator> getShuntCompensatorStream() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public boolean isInMainConnectedComponent() {
+        return getDelegate().isInMainConnectedComponent();
```

Comment:
Related to the comment about connected/synchronous components

Reply:
This will be done later, to be reverted